core/lib/dal/src/models/storage_event.rs

```diff
@@ -67,10 +67,10 @@ pub struct StorageL2ToL1Log {
     pub value: Vec<u8>,
 }
 
-impl From<StorageL2ToL1Log> for L2ToL1Log {
-    fn from(log: StorageL2ToL1Log) -> L2ToL1Log {
-        L2ToL1Log {
-            block_hash: log.block_hash.map(|hash| H256::from_slice(&hash)),
+impl From<StorageL2ToL1Log> for api::L2ToL1Log {
+    fn from(log: StorageL2ToL1Log) -> api::L2ToL1Log {
+        api::L2ToL1Log {
+            block_hash: None,
```

Comment:
Dumb question: Is this field supposed to not be filled unless logs are are returned as a part of [transaction receipts](https://github.com/matter-labs/zksync-era/blob/298a97e800b4c156628050789de7a490a7565d60/core/lib/dal/src/transactions_web3_dal.rs#L132)?

Reply:
AFAIU - yes.

Reply:
Looking into this specific case...`l2_to_l1_logs` table does not have `block_hash` column. and in `dal` we do `NULL::bytea AS "block_hash"` for it

Reply:
Yeah, I get that the current implementation in `main` sets `block_hash` to `null`; my question is whether it's a value that API clients expect. A potential alternative here would be to set it to an L2 block hash that the log is a part of.

Reply:
@perekopskiy  wdyt?

Reply:
Ah, just checked. `api::L2toL1Log` is used only in `get_l2_to_l1_logs_by_hashes`.  In current `get_l2_to_l1_logs_by_hashes` impl  we can not get `block_hash`(we can not do join with miniblocks table but it will be suboptimal) so it is fine to leave everything as is.